hatchways/team-granite
# Image Upload and profile edit integration with backend.

### What this PR does (required):
- Completes the integration with the front end and the back end.
- Added additional routes added to ease integration.
- Live update of the profile photo.

-NEW Upload Photo Dialog

### Screenshots / Videos (front-end only):
- Old Upload profile photo
![PR4](https://user-images.githubusercontent.com/55607809/139632654-f6264146-c880-4d63-8940-34743ada06f6.png)
New Upload Profile Photo
![PRImageUpload](https://user-images.githubusercontent.com/55607809/141421824-9f9811c5-f4e9-43b0-b9f5-5ef099df4e98.png)
- Edit your profile.
![PR2](https://user-images.githubusercontent.com/55607809/139632711-753f6c1f-dd4f-4044-862d-b5d4a8f3be70.png)

### Any information needed to test this feature (required):
- Clear your MongoDB documents in case you have an older version of the schema.
- Create a new user and login.
- Click on the avatar to upload a profile photo.
- Click on the extra setting icon and click on edit profile.
- Edit the profile settings and watch the username change on the dashboard. Use the /auth/user Get request to check the new user details.
- Change the password and try to login with the new password.

### Any issues with the current functionality (optional):
- ~~The profile photo dialog is not well designed.~~
